Kbs.Wpf/User/Update/AccountView.xaml.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Navigation;
+using System.Windows.Shapes;
+using Kbs.Business.User;
+using Kbs.Business.Session;
+using Kbs.Data.User;
+using Kbs.Wpf.Components;
+using Kbs.Wpf.User.Login;
+
+namespace Kbs.Wpf.User.Update
+{
+    /// <summary>
+    /// Interaction logic for AccountView.xaml
+    /// </summary>
+    public partial class AccountView : Page
+    {
+        private readonly UserValidator _userValidator = new UserValidator(); 
+        private readonly INavigationManager _navigationManager;
+        private readonly IUserRepository _userRepository = new UserRepository();
+        private readonly UserEntity _sessionUser;
+        private Func<Page> _navigationTarget;
+
+        public AccountViewModel ViewModel => (AccountViewModel)DataContext;
+        public AccountView(INavigationManager navigationManager)
+        {
+            InitializeComponent();
+            this._navigationManager = navigationManager;
+            _navigationTarget = () => new AccountView(_navigationManager); // change target to homescreen
+            _sessionUser = SessionManager.Instance.Current.User;
+            ViewModel.InputEmail = _sessionUser.Email;
+        }
+
+        private void Submit(object sender, RoutedEventArgs e)
+        {
+            EmailErrorMessage.Content = "";
+            PasswordErrorMessage.Text = "";
+            var user = new UserEntity()
+            {
+                Email = EmailInput.Text,
+                Password = PasswordInput.Password
+            };
+            var validationResult = _userValidator.ValidatorForUpdate(user, PasswordConfirmInput.Password, _userRepository);
+
+            if (validationResult.TryGetValue(nameof(user.Email), out string emailMessage))
+            {
+                if (!emailMessage.Contains("verplicht") && !emailMessage.Contains("bestaat"))
+                {
+                    EmailErrorMessage.Content = emailMessage;
+                }
+            }
+            if (validationResult.TryGetValue(nameof(user.Password), out string passwordMessage))
+            {
+                if (!passwordMessage.Contains("verplicht"))
+                {
+                    PasswordErrorMessage.Text = passwordMessage;
+                }
+            }
+
+            if (validationResult.Count > 1)
```

Comment:
The logic behind this block is really confusing. The condition `validationResult.Count > 1` assumes that if there are multiple validation errors, it's safe to continue. 
On top of this, using parts of error messages like "verplicht" is generally bad practice and does not show the case of what the error is. This makes reading your code difficult.
```suggestion
            if (!validationResult.ContainsKey(nameof(user.Email)) && !validationResult.ContainsKey(nameof(user.Password)))
{
    bool emailUpdated = !string.IsNullOrEmpty(user.Email) && !user.Email.Equals(_sessionUser.Email);
    bool passwordUpdated = !string.IsNullOrEmpty(user.Password) && !user.Password.Equals(_sessionUser.Password);
    
    if (emailUpdated || passwordUpdated)
    {
        string successMessage = "Wijzigingen zijn succesvol aangepast.";
        if (emailUpdated && passwordUpdated)
        {
            successMessage = "Email en Wachtwoord zijn succesvol aangepast.";
        }
        else if (emailUpdated)
        {
            successMessage = "Email is succesvol aangepast.";
        }
        else if (passwordUpdated)
        {
            successMessage = "Wachtwoord is succesvol aangepast.";
        }
        
        MessageBox.Show(successMessage);
        _navigationManager.Navigate(_navigationTarget);
    }
    return;
}
else
{
    MessageBox.Show("Er zijn foutmeldingen. Corrigeer de aangegeven velden.");
}
```